test/unit/controllers/declaration/ConsignmentReferencesControllerSpec.scala

```diff
@@ -63,22 +64,56 @@ class ConsignmentReferencesControllerSpec extends ControllerSpec {
       }
     }
 
-    "return 400 (BAD_REQUEST)" in new SetUp {
+    "return 400 (BAD_REQUEST)" when {
+      "no UCR is submitted" in new SetUp {
 
-      val incorrectForm = Json.toJson(ConsignmentReferences(Some(Ducr("1234")), ""))
+        val incorrectForm = Json.toJson(ConsignmentReferences(Ducr("1234"), ""))
 
-      val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
+        val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
 
-      status(result) must be(BAD_REQUEST)
+        status(result) must be(BAD_REQUEST)
+      }
+
+      "UCR length has been exceeded" in new SetUp {
+
+        val incorrectForm =
+          Json.toJson(ConsignmentReferences(Ducr("1234"), "", Some(TestHelper.createRandomAlphanumericString(36))))
+
+        val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
+
+        status(result) must be(BAD_REQUEST)
```

Comment:
this is more form spec rather then controller

Reply:
The Form/View Spec is already checking for the personal UCR validation. 
This test is just to assert that the HTTP response code is 400 and not 422, 409 or any other.
A response code needs to be asserted in the controller.